### PR TITLE
Only create ThreadGroups if FailOnTimeout.lookForStuckThread is true.

### PR DESCRIPTION
**Pull request by kcooney**
This reduces the differences (where possible) when tests are run with a timeout.
Creating a ThreadGroup can cause noticeable differences, for example in code
that uses java.beans.ThreadGroupContext.

### CODE REVIEW

File: src/main/java/org/junit/internal/runners/statements/FailOnTimeout.java [hunk 1]
```diff
@@ -120,13 +120,24 @@ public FailOnTimeout build(Statement statement) {
...
+              // destroyed after the time-limited thread completes. By ensuring the
+              // ThreadGroup is destroyed, any data associated with the ThreadGroup
+              // (ex: via java.beans.ThreadGroupContext) is destroyed.
+              try {
+                  threadGroup.setDaemon(true);
+              } catch (SecurityException e) {
+                  // Swallow the exception to keep the same behavior as in JUnit 4.12.
+              }
+          }
         }
```
Comment (marcphilipp) [inline src/main/java/org/junit/internal/runners/statements/FailOnTimeout.java:141]: Could we extract this into a method, please?

**Review by marcphilipp — APPROVED** [review]
Good idea! 👍

**Review by stefanbirkner — APPROVED** [review]
Nice catch and thanks for explaining the details in the comments.

File: src/main/java/org/junit/internal/runners/statements/FailOnTimeout.java (hunk 1 above)
Comment (kcooney) [reply src/main/java/org/junit/internal/runners/statements/FailOnTimeout.java]: Done.

**Review by marcphilipp — APPROVED** [review]
Thanks! 👍

**Review by stefanbirkner — APPROVED** [review]

**Comment by kcooney** [comment]
I updated this pull so that the release notes documents the change. @marcphilipp and/or @stefanbirkner would you please review the release notes?

Thanks!

**Review by marcphilipp — APPROVED** [review]

**Review by stefanbirkner — APPROVED** [review]

**Review by blaquezzz — APPROVED** [review]

**Pull request merged**